### PR TITLE
Add class to interact with a Docker Registry server

### PR DESCRIPTION
**Pull request by flavio**
Add Registry class. This can be used to fetch additional information
from a Docker registry v2 using its API.

Right now the Registry class can authenticate against the registry and
fetch the manifest of a Docker image available on the it.

### CODE REVIEW

File: app/models/registry.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env ruby
```
Comment (kalabiyau) [inline app/models/registry.rb]: I think as that probably will be a gem at some point - it is better to settle it in the vendor

File: app/models/registry.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      fail ManifestNotFoundError, "Cannot find manifest for #{repository}:#{tag}"
+    else
+      fail "Something went wrong while fetching manifest for #{repository}:#{tag}:" \
+        "[#{res.code}] - #{res.body}"
+    end
+  end
+
+  def get_request(path, request_auth_token=true)
+    uri = URI.join(@base_url, path)
+    req = Net::HTTP::Get.new(uri)
```
Comment (kalabiyau) [inline app/models/registry.rb]: You can sit on top of RestClient for ruby which gives you all the goodies out of the box

File: app/models/registry.rb (hunk 1 above)
Comment (flavio) [inline app/models/registry.rb]: Yeah, I thought that as well.

Thanks for the catch, I forgot to remove these lines.

File: app/models/registry.rb (hunk 2 above)
Comment (flavio) [inline app/models/registry.rb]: I tried to use it, unfortunately I wasn't able to get the headers out of the first unauthorized error response...

File: app/models/registry.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  def request_auth_token(unhauthorized_response)
+    bearer_realm, query = parse_unhauthorized_response(unhauthorized_response)
+
+    uri = URI("#{bearer_realm}?#{query.to_query}")
+
+    req = Net::HTTP::Get.new(uri)
+    req.basic_auth(@username, @password) if has_credentials?
+
+    res = Net::HTTP.start(
```
Comment (kalabiyau) [inline app/models/registry.rb]: Same here better to use solid solution as this is a golden stardard in Ruby more or less

**Comment by kalabiyau** [comment]
@flavio anyhow, those are just nitpicks, other than coverage below 100% it is good to go, IMO

**Pull request merged**